sympy/core/equation.py

```diff
@@ -0,0 +1,342 @@
+as students are taught to do when attempting to isolate (solve for) a
+variable. Thus the statement ``Equation/b`` yields a new equation
+``Equation.lhs/b = Equation.rhs/b``
+
+The intent is to allow using the mathematical tools in SymPy to rearrange
+equations and perform algebra in a stepwise fashion. In this way more people
+can successfully perform algebraic rearrangements without stumbling over
+missed details such as a negative sign. This mimics the capabilities available
+in [SageMath](https://www.sagemath.org/) and
+[Maxima](http://maxima.sourceforge.net/).
+"""
+
+from .basic import Basic
+from .sympify import _sympify
+
+
+class Equation(Basic):
+    """
+    This class defines an equation with a left-hand-side (lhs) and a right-
+    hand-side (rhs) connected by the "=" operator (e.g. $p*V = n*R*T$).
+
+    Explanation
+    ===========
+    This class defines relations that all high school and college students
+    would recognize as mathematical equations. At present only the "=" relation
+    operator is recognized.
+
+    This class is intended to allow using the mathematical tools in SymPy to
+    rearrange equations and perform algebra in a stepwise fashion. In this
+    way more people can successfully perform algebraic rearrangements without
+    stumbling over missed details such as a negative sign.
+
+    Create an equation with the call ``Equation(lhs,rhs)``, where ``lhs`` and
+    ``rhs`` are any valid Sympy expression. ``Eqn(...)`` is a synonym for
+    ``Equation(...)``.
+
+    Examples
+    ========
+    >>> from sympy import var, Equation, Eqn, exp, log, integrate, Integral
+    >>> from sympy import solve
+    >>> a, b, c = var('a b c')
+    >>> str(Equation(a,b/c))
+    a=b/c
+    >>> t=Eqn(a,b/c)
+    >>> str(t)
+    a=b/c
+    >>> str(t*c)
+    a*c=b
+    >>> str(c*t)
+    a*c=b
+    >>> str(exp(t))
+    exp(a)=exp(b/c)
+    >>> str(exp(log(t)))
+    a=b/c
+
+    Integration can only be performed on one side at a time.
+    >>> q=Eqn(a*c,b/c)
+    >>> integrate(q,b,side='rhs')
+    b**2/(2*c)
+    >>> integrate(q,b,side='lhs')
+    a*b*c
+
+    Make a pretty statement of integration from an equation
+    >>> str(Eqn(Integral(q.lhs,b),integrate(q,b,side='rhs')))
+    Integral(a*c, b)=b**2/(2*c)
+
+    This is duplicated by the convenience function self.integ
+    >>> str(q.integ(b))
+    Integral(a*c, b)=b**2/(2*c)
+
+    SymPy's solvers do not understand these equations. They expect an
+    expression that the solver assumes = 0. Thus to use the solver the
+    equation must be rearranged so that all non-zero symbols are on one side.
+    Then just the non-zero symbolic side is passed to ``solve()``.
+    >>> t2 = t-t.rhs
+    >>> str(t2)
+    a-b/c=0
+    >>> solve(t2.lhs,c)
+    [b/a]
+    """
+
+    def __new__(cls, lhs, rhs, **kwargs):
+        check = kwargs.pop('check', True)
+        lhs = _sympify(lhs)
+        rhs = _sympify(rhs)
```

Comment:
I think it should be checked (always) that the lhs and rhs are `Expr` right here after sympification. The operations that this class defines (unlike those of Eq) are specific to `Expr` so we should make sure that this is only usable with `Expr`. With `Equality` we can do things like:
```julia
In [1]: Eq(FiniteSet(x), FiniteSet(y, z))                                                                                                                     
Out[1]: {x} = {y, z}

In [2]: Eq(FiniteSet(x), FiniteSet(y, z)).subs({x:1, y:1, z:1})                                                                                               
Out[2]: True
```
That's useful but I don't think we want it here so `Equation(nonExpr1, nonExpr2)` should just give a `TypeError`.

Reply:
Makes sense.

Reply:
I would also consider to make `rhs` optional and check if `lhs` is an instance of `Relational`; in that case, then `lhs, rhs = lhs.args`. This, together with the method `as_Boolean`, allows for a faster back-and-forth conversion.